### ISSUE TITLE
 Training with examples into multiple TFRecord files (shards).

### ISSUE THREAD

**Issue by aguirrejuan** [zzh8829/yolov3-tf2#301]
Hi!, first of all thanks for your code.

I am trying to train with the data set [MOT20](https://motchallenge.net/data/MOT20/) which I have divided into TFrecord each with 200 images and pass it to the model as shown below

Directory : 
../input/mot20-tfrecords/
└── MOT20_tfrecords_3
    ├── MOT20_train_00000-of-00003_ 00000-of-00002.records
    ├── MOT20_train_00000-of-00003_ 00001-of-00002.records
    ├── MOT20_train_00001-of-00003_ 00000-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00001-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00002-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00003-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00004-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00005-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00006-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00007-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00008-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00009-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00010-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00011-of-00013.records
    ├── MOT20_train_00001-of-00003_ 00012-of-00013.records
    ├── MOT20_train_00002-of-00003_ 00000-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00001-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00002-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00003-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00004-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00005-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00006-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00007-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00008-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00009-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00010-of-00012.records
    ├── MOT20_train_00002-of-00003_ 00011-of-00012.records
    ├── MOT20_train_00003-of-00003_ 00000-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00001-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00002-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00003-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00004-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00005-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00006-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00007-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00008-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00009-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00010-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00011-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00012-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00013-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00014-of-00016.records
    ├── MOT20_train_00003-of-00003_ 00015-of-00016.records
    ├── MOT20_val_00000-of-00003.records
    ├── MOT20_val_00001-of-00003.records
    ├── MOT20_val_00002-of-00003.records
    └── MOT20_val_00003-of-00003.records

 python train.py 
	--dataset ./input/mot20-tfrecords/MOT20_tfrecords_3/MOT20_train_\*-of-\*_\*-of-\*.records 
	--val_dataset ./input/mot20-tfrecords/MOT20_tfrecords_3/MOT20_val_\*-of-\*.records \
	--classes ./data/MOT2020.names \
	--num_classes 1 \
	--mode fit --transfer darknet \
	--batch_size 32 \
	--epochs 1000 \
	--weights /kaggle/input/weights/yolov3.tf \
	--weights_num_classes 80 \ 

But I get the following training history
Epoch 4/1000
7/7 [==============================] - 16s 2s/step - loss: 663.5157 - yolo_output_0_loss: 73.1910 - yolo_output_1_loss: 80.5648 - yolo_output_2_loss: 498.9141 - val_loss: 12106.3467 - val_yolo_output_0_loss: 1537.6014 - val_yolo_output_1_loss: 9196.6729 - val_yolo_output_2_loss: 1361.2030 - lr: 0.0010 

I'm not sure if I'm really going through all the images every epoch. If you look at time step = 7, so it is like 200/32 and it should be 800/32, could you help me understand what happened?

**Comment by aguirrejuan** [comment]
hi! the solution is the next

python train.py
--dataset "./input/mot20-tfrecords/MOT20_tfrecords_3/MOT20_train_*-of-**-of-*.records"
--val_dataset "./input/mot20-tfrecords/MOT20_tfrecords_3/MOT20_val*-of-*.records"
--classes ./data/MOT2020.names
--num_classes 1
--mode fit --transfer darknet
--batch_size 32
--epochs 1000
--weights /kaggle/input/weights/yolov3.tf
--weights_num_classes 80 \

:)
